pkg/metrics/sender.go

```diff
@@ -0,0 +1,70 @@
+package metrics
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/replicatedhq/helmvm/pkg/addons/adminconsole"
+)
+
+// Send is a helper function that sends an event to the metrics endpoint.
+// Metrics endpoint can be overwritten by the license.spec.endpoint field
+// or by the HELMVM_METRICS_BASEURL environment variable, the latter has
+// precedence over the former.
+func Send(ctx context.Context, ev Event) {
+	var baseURL = "https://replicated.app"
+	var custom adminconsole.AdminConsoleCustomization
+	if license, err := custom.License(); err != nil {
+		logrus.Warnf("unable to read license: %s", err)
+		return
+	} else if license != nil && license.Spec.Endpoint != "" {
+		baseURL = license.Spec.Endpoint
+	}
+	if os.Getenv("HELMVM_METRICS_BASEURL") != "" {
+		baseURL = os.Getenv("HELMVM_METRICS_BASEURL")
+	}
+	sender := Sender{baseURL}
+	sender.Send(ctx, ev)
+}
+
+// Sender sends events to the metrics endpoint.
+type Sender struct {
+	baseURL string
+}
+
+// Send sends an event to the metrics endpoint.
+func (s *Sender) Send(ctx context.Context, ev Event) {
+	url := fmt.Sprintf("%s/helmbin_metrics/%s", s.baseURL, ev.Title())
+	payload, err := s.payload(ev)
+	if err != nil {
+		logrus.Infof("unable to get payload for event %s: %s", ev.Title(), err)
```

Comment:
Why is this info and the rest level debug?

Reply:
my bad, it should have been a debugf as well. fixed.